src/Stratis.Bitcoin/Utilities/MemorySizeCache.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections.Generic;
+
+namespace Stratis.Bitcoin.Utilities
+{
+    /// <summary>
+    /// Memory cache that implements the Least Recently Used (LRU) policy.
+    /// </summary>
+    public class MemorySizeCache<TKey, TValue>
```

Comment:
80-90% of the class was copypasted from MemoryCountCache
Please inherit from it and override what is needed in order to avoid copy-pasting things.